compiler/src/codegen/compcore.re

```diff
@@ -1015,45 +958,233 @@ let compile_imm =
   | MImmTrap => Expression.Unreachable.make(wasm_mod)
   };
 
-let call_error_handler = (wasm_mod, env, err, args) => {
-  let (err_value_ident, err_func_ident) =
-    switch (err) {
-    | Runtime_errors.MatchFailure => (match_failure_ident, None)
-    | Runtime_errors.IndexOutOfBounds => (index_out_of_bounds_ident, None)
-    | Runtime_errors.IndexNonInteger => (index_non_integer_ident, None)
-    | Runtime_errors.AssertionError => (
-        assertion_error_closure_ident,
-        Some(get_wasm_imported_name(exception_mod, assertion_error_ident)),
+/** Heap allocations. */
```

Comment:
None of this has changed, it's just moved higher so we can do allocations for exceptions (which was previously just calling a function).